draft-ietf-mls-protocol.md

```diff
@@ -1848,9 +1848,14 @@ enum {
     (255)
 } SenderType;
 
+struct {
+    opaque sender_identity;
+    opaque sender_endpoint_id;
+} Client
```

Comment:
The bloat induced by this makes me pretty sad, and relying on just these two fields seems a little brittle.

As an alternative, what about using a hash of the leaf KeyPackage?:
* We only care about exact equality of these things, so we could use a hash of these values.  That also has the benefit of revealing less information on the wire.
* We only care about populated leaves, which contain a KeyPackage
* We already need `KeyPackage.hash()` for purposes of building/processing Welcome messages

That would get you down to the more compact:

```
uint8 KeyPackageID<0..255>;
```

I would be slightly happier if it were a fixed-length value and shorter than a hash output.  It's tempting to propose truncating the hash down to 128 bits, which seems unlikely to result in collisions in any practical scenario.  If it is fixed-length, you could have a nice compact value:

```
uint8 KeyPackageID[16];
```

Reply:
Note also that if we make a smaller-than-hash-KeyPackage-ID thing here, we should also use it in Welcome.

Reply:
I like the idea, although I'm always a bit skeptical when it comes to shortening hashes. In this case it doesn't serve much of a security purpose, but I'd still prefer the full hash.

Reply:
Any suggestions for a name, though? The KeyPackage hash does not really represent the client anymore, but rather the client's presence in this particular group.

Reply:
I kind of like `KeyPackageID`, as above.